File: scala-example/spring-boot-example/src/main/scala/com/github/poad/scala/springboot/example/service/MessageService.scala
package com.github.poad.scala.springboot.example.service

import java.util.Optional

import com.github.poad.scala.springboot.example.content.Message
import org.springframework.stereotype.Service

@Service
trait MessageService {
  def all(): Seq[Message]

  def byId(id: String): Optional[Message]

  def create(message: Message): String

  def update(id: String, message: Message): Optional[Message]

  def deleteById(id: String): Optional[Message]
}

File: java-maven-example/spring-boot-minimal-jdk/src/main/java/com/github/poad/example/service/MessageService.java
package com.github.poad.example.service;

import com.github.poad.example.domain.Message;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface MessageService {
    List<Message> findAll();

    Optional<Message> findById(String id);

    Message create(String message);

    Optional<Message> update(String id, String message);

    void delete(String id);
}
